lib/defaults/server-metadata.js

```diff
@@ -5,6 +5,7 @@ const serviceResponseDefaults = {
   hasVersionedData: false,
   supportsDisconnectedEditing: false,
   supportedQueryFormats: 'JSON',
+  supportsRelationshipsResource: true,
```

Comment:
The default for this should not be `true`.  It can only be true if the provider you are using supports the `queryRelatedRecords` method and server the special JSON noted in the PR description when this method comes through.

Reply:
I think the reality of clients are that they don't look at this value at all but only at if relationships exist, but putting together an update to clean this up.